### PR TITLE
FIX: correctly untrack topics when dismiss unread

### PR DESCRIPTION
**Pull request by lis2**
Bug introduced when dismiss modal was moved to new component-based API - https://github.com/discourse/discourse/pull/22262

We need to track `dismissTopics` property to send correct request to  server.

Meta: https://meta.discourse.org/t/dismiss-all-unread/280948

Demo

https://github.com/discourse/discourse/assets/72780/6f2881b0-f37d-41c5-bad5-c51c8219274b

### CODE REVIEW

File: app/assets/javascripts/discourse/app/components/modal/dismiss-read.hbs [hunk 1]
```diff
@@ -8,6 +8,7 @@
       <PreferenceCheckbox
         @labelKey="topics.bulk.also_dismiss_topics"
         @checked={{this.dismissTopics}}
+        class="stop-tracking"
```
Comment (martin-brennan) [inline app/assets/javascripts/discourse/app/components/modal/dismiss-read.hbs:11]:
```suggestion
        class="dismiss-read-modal__stop-tracking"
```

Minor, but we should be following BEM rules when making CSS classes

Comment (lis2) [reply app/assets/javascripts/discourse/app/components/modal/dismiss-read.hbs]: Nice one, thank you 💯

File: spec/system/page_objects/components/topic_view.rb [hunk 2]
```diff
@@ -10,6 +10,10 @@ def has_read_post?(post)
           wait: Capybara.default_max_wait_time * 2,
         )
       end
+
+      def has_tracking_status?(name)
```
Comment (martin-brennan) [inline spec/system/page_objects/components/topic_view.rb:14]: This is too vague, this applies to tons of elements on the page. We have SelectKit page object helpers, you should use those here. https://github.com/discourse/discourse/blob/8f78425e0cfba073b3b7c373515b9adac2ce8d41/spec/system/page_objects/components/select_kit.rb . And an example:

https://github.com/discourse/discourse/blob/8f78425e0cfba073b3b7c373515b9adac2ce8d41/spec/system/composer/default_to_subcategory_spec.rb#L18-L22

File: spec/system/dismissing_new_spec.rb [hunk 3]
```diff
@@ -33,6 +34,24 @@
...
+        sign_in(user)
+
+        visit("/t/#{topic.id}")
+
+        expect(topic_view).to have_tracking_status("tracking")
+      end
+
+      topic_list_controls.dismiss_unread(untrack: true)
+
+      using_session(:tab_1) { expect(topic_view).to have_tracking_status("regular") }
```
Comment (martin-brennan) [inline spec/system/dismissing_new_spec.rb:53]: I'm surprised this is updated via message bus, nice!

**Review by martin-brennan — APPROVED** [review]
Nice, just a minor spec fix to do :)

**Pull request merged**